src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1057,13 +1057,12 @@ private void AddValueInternal(DateTime time, Span<double> values, Span<byte> tag
                     var segmentLastTimestamp = segment.GetLastTimestamp(BaselineDate);
                     if (segmentLastTimestamp == BaselineDate && time == segmentLastTimestamp) // all the entries in the segment has the same timestamp
                     {
-                        if (FromReplication == false)
-                            throw new InvalidDataException(
-                                $"Segment reached to capacity and cannot receive more values with {time} timestamp on time series {_name} for {_docId} . " +
-                                "You may choose a different timestamp.");
+                        if (FromReplication)
+                            RaiseAlertOnMaxCapacityForSingleSegmentReached(values);
 
-                        DiscardsAllEntriesAtSameTimestampToSingleDeadEntry(ref segment, tagSlice, values);
```

Comment:
Why we don't call `DiscardsAllEntriesAtSameTimestampToSingleDeadEntry` anymore?

Reply:
No, the current logic has a bug so we decided to handle this at later stage (https://issues.hibernatingrhinos.com/issue/RavenDB-17557)